### PR TITLE
Introduce SQS client factory and S3 factory

### LINKED ISSUE #129: Setting the credentials in production
Whats the best way to set up the credentials in non-aws production environment? Setting the access and secret key environmental variable is not ideal.

### PR DESCRIPTION
**Pull request by danielmarbach**
Closes #129

Introduces a possibility to specify a client factory for the SQS and the S3 client creation. 

## Documentation

https://github.com/Particular/docs.particular.net/pull/3337

## SQS Client Factory

Example

```
transport.ClientFactory(() => new AmazonSQSClient(
   new AmazonSQSConfig {
      RegionEndpoint = "ap-southeast-2"
   });
```

## S3 Client Factory

Example

```
transport.ClientFactory(() => new AmazonS3Client(
   new AmazonS3Config {
      RegionEndpoint = "ap-southeast-2"
   });
```

### CODE REVIEW

**Comment by mauroservienti** [comment]
Given all the magic the the AWS client put in place to support many configuration sources, why enforcing people on providing a factory that probably in the majority of cases would be something like:

```
transportSettings.ClientFactory(() => new AmazonSQSClient());
```

I'd rather make the above the default behavior, behavior the if needed can be easily overwritten by users.

**Comment by danielmarbach** [comment]
@mauroservienti that would be good for me too let's wait for @SimonCropp to chime in.

**Comment by SimonCropp** [comment]
> I'd rather make the above the default behavior, behavior the if needed can be easily overwritten by users.

@mauroservienti yep that sounds reasonable

File: src/AcceptanceTests/SetupFixture.cs [hunk 1]
```diff
@@ -35,17 +35,19 @@ public void OneTimeSetUp()
...
-                catch (Exception ex)
-                {
-                    Console.WriteLine($"Exception when deleting queue: {ex}");
+                    try
+                    {
+                        await sqsClient.DeleteQueueAsync(queueUrl).ConfigureAwait(false);
+                    }
+                    catch (Exception ex)
+                    {
+                        Console.WriteLine($"Exception when deleting queue: {ex}");
```
Comment (SimonCropp) [inline src/AcceptanceTests/SetupFixture.cs]: why Console.WriteLine  ?

Comment (danielmarbach) [reply src/AcceptanceTests/SetupFixture.cs]: This was already there, I think it is done because then it is visible on TC in the build logs

**Comment by danielmarbach** [comment]
Added the [doc PR WIP](https://github.com/Particular/docs.particular.net/pull/3337) to the description

**Comment by danielmarbach** [comment]
@mauroservienti @SimonCropp as soon as the PR is merged I will continue with the doc PR.

File: src/NServiceBus.AmazonSQS/Configure/S3Settings.cs [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    using System;
+    using System.Linq;
+    using Amazon.S3;
+    using Configuration.AdvancedExtensibility;
+    using Settings;
+
+    /// <summary>
+    /// 
+    /// </summary>
+    public class S3Settings : ExposeSettings
```
Comment (SimonCropp) [inline src/NServiceBus.AmazonSQS/Configure/S3Settings.cs]: needs some xml comments.

**Pull request merged**